### PR TITLE
SYN-5701: CVSS norm doesn't handle unordered vectors

### PR DESCRIPTION
**Pull request by blackout-vtx**
- Update risk model so risk:vuln:cvss:v2 and :cvss:v3 props are proper types that can do normalization and validation of values
- Added migration model to synapse-regression repo
- Updated tests

Resolves SYN-5701

### CODE REVIEW

File: synapse/models/risk.py [hunk 1]
```diff
@@ -1,14 +1,41 @@
...
+import synapse.lib.chop as s_chop
+import synapse.lib.types as s_types
 import synapse.lib.module as s_module
 
-cvss_v2_re = r'AV:(L|A|N)\/AC:(H|M|L)\/Au:(M|S|N)\/C:(N|P|C)\/I:(N|P|C)\/A:(N|P|C)(\/E:(ND|U|POC|F|H)\/RL:(ND|OF|TF|W|U)\/RC:(ND|UC|UR|C))?(\/CDP:(ND|N|L|LM|MH|H)\/TD:(ND|N|L|M|H)\/CR:(ND|L|M|H)\/IR:(ND|L|M|H)\/AR:(ND|L|M|H))?'
 
-cvss_v3_re = r'AV:(N|A|L|P)\/AC:(L|H)\/PR:(N|L|H)\/UI:(N|R)\/S:(U|C)\/C:(N|L|H)\/I:(N|L|H)\/A:(N|L|H)(\/E:(X|U|P|F|H)\/RL:(X|O|T|W|U)\/RC:(X|U|R|C))?(\/CR:(X|L|M|H)\/IR:(X|L|M|H)\/AR:(X|L|M|H)\/MAV:(X|N|A|L|P)\/MAC:(X|L|H)\/MPR:(X|N|L|H)\/MUI:(X|N|R)\/MS:(X|U|C)\/MC:(X|N|L|H)\/MI:(X|N|L|H)\/MA:(X|N|L|H))?'
+class CvssV2(s_types.Str):
+
+    def postTypeInit(self):
```
Comment (vEpiphyte) [inline synapse/models/risk.py:10]: do we need this? If we're subclassing s_types.str we should be able to just override `_normPyStr`  ?

File: synapse/models/risk.py [hunk 2]
```diff
@@ -1,14 +1,33 @@
...
+import synapse.lib.types as s_types
 import synapse.lib.module as s_module
 
-cvss_v2_re = r'AV:(L|A|N)\/AC:(H|M|L)\/Au:(M|S|N)\/C:(N|P|C)\/I:(N|P|C)\/A:(N|P|C)(\/E:(ND|U|POC|F|H)\/RL:(ND|OF|TF|W|U)\/RC:(ND|UC|UR|C))?(\/CDP:(ND|N|L|LM|MH|H)\/TD:(ND|N|L|M|H)\/CR:(ND|L|M|H)\/IR:(ND|L|M|H)\/AR:(ND|L|M|H))?'
 
-cvss_v3_re = r'AV:(N|A|L|P)\/AC:(L|H)\/PR:(N|L|H)\/UI:(N|R)\/S:(U|C)\/C:(N|L|H)\/I:(N|L|H)\/A:(N|L|H)(\/E:(X|U|P|F|H)\/RL:(X|O|T|W|U)\/RC:(X|U|R|C))?(\/CR:(X|L|M|H)\/IR:(X|L|M|H)\/AR:(X|L|M|H)\/MAV:(X|N|A|L|P)\/MAC:(X|L|H)\/MPR:(X|N|L|H)\/MUI:(X|N|R)\/MS:(X|U|C)\/MC:(X|N|L|H)\/MI:(X|N|L|H)\/MA:(X|N|L|H))?'
+class CvssV2(s_types.Str):
+
+    def _normPyStr(self, text):
+        return s_chop.cvss2_normalize(text), {}
```
Comment (vEpiphyte) [inline synapse/models/risk.py:11]: Can we catch the BadDataValu exceptions thrown by the normalize functions and raise BadTypeValu's instead? That would be more consistent with other types.

Comment (invisig0th) [reply synapse/models/risk.py:11]: BadTypeValu being raised is also required for conditional assignment to work :+1:

Comment (invisig0th) [reply synapse/models/risk.py:11]: ( we could also just tweak the normalize function to raise BadTypeValu

File: synapse/models/risk.py [hunk 3]
```diff
@@ -1,14 +1,43 @@
...
-cvss_v3_re = r'AV:(N|A|L|P)\/AC:(L|H)\/PR:(N|L|H)\/UI:(N|R)\/S:(U|C)\/C:(N|L|H)\/I:(N|L|H)\/A:(N|L|H)(\/E:(X|U|P|F|H)\/RL:(X|O|T|W|U)\/RC:(X|U|R|C))?(\/CR:(X|L|M|H)\/IR:(X|L|M|H)\/AR:(X|L|M|H)\/MAV:(X|N|A|L|P)\/MAC:(X|L|H)\/MPR:(X|N|L|H)\/MUI:(X|N|R)\/MS:(X|U|C)\/MC:(X|N|L|H)\/MI:(X|N|L|H)\/MA:(X|N|L|H))?'
+class CvssV2(s_types.Str):
+
+    def _normPyStr(self, text):
+        try:
+            return s_chop.cvss2_normalize(text), {}
+
+        except s_exc.BadDataValu as exc:
+            mesg = exc.get('mesg')
+            raise s_exc.BadTypeValu(mesg=mesg) from None
```
Comment (vEpiphyte) [inline synapse/models/risk.py:16]: can we add `name=self.name` so we capture the name of the property/form as well?

Comment (blackout-vtx) [reply synapse/models/risk.py]: I removed this and changed the BadDataValu's to BadTypeValu in chop

**Review by invisig0th — DISMISSED** [review]

**Review by vEpiphyte — APPROVED** [review]

**Pull request merged**